File: frontend/src/services/listeningSentenceFeedbackService.js
// src/services/listeningSentenceFeedbackService.js
import axiosClient from "../api/axiosClient";

const listeningSentenceFeedbackService = {
  create: (data) => {
    return axiosClient.post("/v1/listening-sentence-feedbacks", data);
  },

  update: (feedbackId, data) => {
    return axiosClient.put(
      `/v1/listening-sentence-feedbacks/${feedbackId}`,
      data,
    );
  },

  delete: (feedbackId) => {
    return axiosClient.delete(`/v1/listening-sentence-feedbacks/${feedbackId}`);
  },

  getMyFeedbackBySentence: (sentenceId) => {
    return axiosClient.get(
      `/v1/listening-sentence-feedbacks/sentence/${sentenceId}/my`,
    );
  },

  getMyFeedbacks: () => {
    return axiosClient.get("/v1/listening-sentence-feedbacks/my");
  },

  getMyFeedbacksByLesson: (lessonId) => {
    return axiosClient.get(
      `/v1/listening-sentence-feedbacks/my/lesson/${lessonId}`,
    );
  },


  getBySentence: (sentenceId) => {
    return axiosClient.get(
      `/v1/listening-sentence-feedbacks/sentence/${sentenceId}`,
    );
  },

  getByLesson: (lessonId) => {
    return axiosClient.get(
      `/v1/listening-sentence-feedbacks/lesson/${lessonId}`,
    );
  },
};

export default listeningSentenceFeedbackService;

File: backend/src/main/java/com/englishlearning/backend/service/ListeningSentenceFeedbackService.java
package com.englishlearning.backend.service;

import com.englishlearning.backend.dto.request.ListeningSentenceFeedbackRequest;
import com.englishlearning.backend.dto.response.ListeningSentenceFeedbackResponse;

import java.util.List;

public interface ListeningSentenceFeedbackService {


    ListeningSentenceFeedbackResponse create(Long userId, ListeningSentenceFeedbackRequest request);
    ListeningSentenceFeedbackResponse update(Long userId, Long feedbackId, ListeningSentenceFeedbackRequest request);
    void delete(Long userId, Long feedbackId);
    ListeningSentenceFeedbackResponse getByStudentAndSentence(Long userId, Long sentenceId);
    List<ListeningSentenceFeedbackResponse> getMyFeedbacks(Long userId);
    List<ListeningSentenceFeedbackResponse> getBySentence(Long sentenceId);
    List<ListeningSentenceFeedbackResponse> getByLesson(Long lessonId);
    List<ListeningSentenceFeedbackResponse> getMyFeedbacksByLesson(Long userId, Long lessonId);
}